blackduck/ClientCore.py

```diff
@@ -35,12 +35,13 @@ def _request(
     """
 
     headers = {
-        'accept' : 'application/json'
+        'accept': 'application/json',
+        'content-type': 'application/json'
```

Comment:
Overall, the endpoints are very inconsistent to how they hand the 'accept' and/or 'content-type' media headers. For example while most accept either, some only accept one or the other.

These defaults, the same as HubInstance, work for the majority but ideally the developer should specify them explicitly if they want to ensure consistent results and detect breakage during Hub upgrades.

Reply:
In general, **application/json** means _latest schema version_ just FYI. All the endpoints _should_ take application/json although I know of a couple of exceptions. Of course, to pin the schema and avoid backwards compatibility issues it is highly recommended to choose a (public) media type from the API docs. 

One of the _mistakes_ I made in earlier versions was to hard-code media types into the methods. I believe there should be a default (suggest application/json, aka latest) and the calling client should be allowed to override and supply whatever accept, content-type values they want.

Reply:
Agreed. It is one of the uglier warts I encountered when using the REST API. With the new client, it is easy to explicitly set headers if desired. E.g.

headers = {
    'accept': "application/vnd.blackducksoftware.project-detail-4+json",
    'content-type': "application/vnd.blackducksoftware.project-detail-4+json"
}
for project in hub.get_resource('projects', headers=headers):

'application/json' as the default will work for most end-points bar the few exceptions. I will document this later on this week.